src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedInteractiveInitializerMethod.vb

```diff
@@ -155,10 +155,20 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             If useSiteDiagnostic IsNot Nothing Then
                 diagnostics.Add(useSiteDiagnostic, NoLocation.Singleton)
             End If
-            resultType = compilation.GetTypeByReflectionType(submissionReturnType, diagnostics)
+            ' if no explicit return type is set on ScriptCompilationInfo, it will default to
+            ' the _host_ corlib's System.Object (GetType(object)) and should never be null.
+            ' In this case, redirect the host System.Object to the compilation's target
+            ' corlib's System.Object. This allows cross compiling scripts to build on one
+            ' corlib and run on another.
+            ' cf. https://github.com/dotnet/roslyn/issues/8506
+            If submissionReturnType Is GetType(Object) Then
```

Comment:
C# implementation is using user-defined equality operator, this code is using reference equality.
Also, the comment I gave for C# implementation is applicable here, i.e. this place feels like a wrong place for the fix.